admin/server/auth/handlers.go

```diff
@@ -355,3 +357,84 @@ func (a *Authenticator) authLogoutCallback(w http.ResponseWriter, r *http.Reques
 	// Redirect to UI (usually)
 	http.Redirect(w, r, redirect, http.StatusTemporaryRedirect)
 }
+
+// / handleAuthorizeRequest handles the incoming OAuth2 Authorization request
+// and generates an authorization code while associating the code challenge.
+func (a *Authenticator) handleAuthorizeRequest(w http.ResponseWriter, r *http.Request) {
+	claims := GetClaims(r.Context())
+	if claims == nil {
+		internalServerError(w, fmt.Errorf("did not find any claims, %w", errors.New("server error")))
+		return
+	}
+	if claims.OwnerType() == OwnerTypeAnon {
+		// not logged in, redirect to login
+		// TODO how to choose between login and signup?
```

Comment:
Probably we need a way for the caller to indicate a preference (based on whether a login or signup button is pressed). Maybe we could support the `screen_hint=signup` URL arg like Auth0 does? See here: https://github.com/rilldata/rill/blob/b1e5563681aa8bff0894a1eb978527a8d320e623/admin/server/auth/handlers.go#L109

Reply:
Not sure what the UI will do, just thought we might maintain optionality and let the UI decide. But probably you're right it will go into a single flow (and our Auth0 login screen allows switching to signup), so let's just not do this, and see if anyone requests it.